src/rust/engine/graph/src/lib.rs

```diff
@@ -536,14 +531,18 @@ impl<N: Node> Graph<N> {
     if dst_retry {
       // Retry the dst a number of times to handle Node invalidation.
       let context = context.clone();
-      let deadline = Instant::now() + self.invalidation_timeout;
       loop {
         match entry.get(&context, entry_id).await {
           Ok(r) => break Ok(r),
           Err(err) if err == N::Error::invalidated() => {
-            if deadline < Instant::now() {
-              break Err(N::Error::exhausted());
-            }
+            let node = {
+              let inner = self.inner.lock();
+              inner.unsafe_entry_for_id(entry_id).node().clone()
+            };
+            info!(
+              "Filesystem changed during run: retrying `{}` in {:?}...",
```

Comment:
To reduce the risk of truncation:

```suggestion
              "Filesystem changed: retrying `{}` in {:?}...",
```

Reply:
I'm wondering if it would be worth rewording so that the time appears before the node description also. I'm concerned some nodes may be really long.

Reply:
I think it's important to mention "during run", because as you said, this is independent from `--loop`... also, I want to send the signal that this is tied specifically to the thing that is re-running (it is: if a file unrelated to the thing that was running is changed, you won't see this).

Regarding truncation: these messages are not truncated. Perhaps they should be, but.

Reply:
Okay, fair. Thanks